src/Concerns/CapturesState.php

```diff
@@ -110,6 +111,14 @@ public function configureGlobalRequestSampling(): void
      */
     public function configureGlobalCommandSampling(): void
     {
+        if (Compatibility::$contextExists) {
+            if (Context::hasHidden('nightwatch_should_sample')) {
+                $this->sample(Compatibility::getSamplingFromContext() ? 1.0 : 0.0);
+
+                return;
+            }
+        }
+
         $this->sample($this->config['sampling']['commands']);
```

Comment:
It would be nice if we could just use the existing `Compatibility::getSamplingFromContext` method here.

We'll need to make the default parameter nullable.


```suggestion
$this->sample(Compatibility::getSamplingFromContext(default: null) ?? $this->config['sampling']['commands']);
```

I believe we can make the types happy by using a PHPStan @template. Lets chat about this if you aren't sure about that one.

Reply:
Ohh nice I was trying to think of something, but doesn't `getSamplingFromContext` only take and return a boolean?

Reply:
It does but we would adjust the default to accept nullable. We should also be able to use `@template` to ensure the return type is understood by PHPStan, e.g., `bool` or `null|bool` depending on what you pass as the default.

Also, this is what I should have suggested:

```php
$this->sample(match(Compatibility::getSamplingFromContext(default: null)) {
    true => 1.0,
    false => 0.0,
    null => $this->config['sampling']['commands'],
});
```

Reply:
What do you think now? I could use some of your ideas to make the phpstan stuff nicer I think